charts/app-stable-diffusion/templates/ingress.yaml

```
@@ -45,3 +45,51 @@ spec:
           {{- end }}
     {{- end }}
 {{- end }}
+
+{{- if .Values.stablestudio.enabled -}}
+{{- $ssPort := .Values.stablestudio.service.port -}}
+{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.stablestudio.ingress.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.stablestudio.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ include "stablestudio.fullname" . }}
+  labels:
+    {{- include "stablestudio.labels" $ | nindent 4 }}
+  {{- with .Values.stablestudio.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if and .Values.ingress.className (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion) }}
+  ingressClassName: {{ .Values.ingress.className }}
+  {{- end }}
+  {{- if .Values.stablestudio.ingress.tls }}
+  tls:
+    {{- range .Values.stablestudio.ingress.tls }}
+    - hosts:
+        {{- range .hosts }}
+        - {{ . | quote }}
+        {{- end }}
+      secretName: {{ .secretName }}
+    {{- end }}
+  {{- end }}
+  rules:
+    {{- range .Values.stablestudio.ingress.hosts }}
+    - host: {{ .host | quote }}
+      http:
+        paths:
+          {{- range .paths }}
+          - path: {{ .path }}
+            pathType: {{ .pathType }}
+            backend:
+              service:
+                name: {{ include "stablestudio.fullname" $ }}
+                port:
+                  number: {{ $ssPort }}
```

nit, we could stick to port name instead of a number to simplify

```suggestion
                  name: http
```